src/System.Net.NetworkInformation/tests/FunctionalTests/NetworkAvailabilityChangedTests.cs

```diff
@@ -27,6 +27,20 @@ public void NetworkAvailabilityChanged_JustRemove_Success()
             NetworkChange.NetworkAvailabilityChanged -= _availabilityHandler;
         }
 
+        [Fact]
+        [ActiveIssue(33530, TestPlatforms.FreeBSD)]
+        public void NetworkAddressChanged_Add_DoesNotBlock()
+        {
+            // Register without unregistering.
+            // This should not block process exit. If it does, this test will pass
+            // but we would fail to exit test run at the end.
+            // We cannot test this via RemoteInvoke() as that call System.Exit()
+            // and forces quit even when foreground threads are running.
+            NetworkChange.NetworkAddressChanged += _addressHandler;
+            {
+            };
```

Comment:
Hmm.  I know we spoke about it offline, but seeing it now I realize this could negatively interact with other tests.  Do any other tests validate the behavior when going from zero to one handlers, validate the behavior when the last handler is removed, etc.?

Reply:
I did not see any. All the tests are primitive in sense that they only verify that handler can be set but they do not test actual address changes as that is difficult to do in CI.